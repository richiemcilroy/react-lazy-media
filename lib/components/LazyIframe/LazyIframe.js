'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.LazyIframe = void 0;

var _react = _interopRequireWildcard(require('react'));

var _propTypes = _interopRequireDefault(require('prop-types'));

var _classnames = _interopRequireDefault(require('classnames'));

var _ReactLazyMedia = require('../../ReactLazyMedia');

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : {default: obj};
}

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj;
  } else {
    var newObj = {};
    if (obj != null) {
      for (var key in obj) {
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
          var desc =
            Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {};
          if (desc.get || desc.set) {
            Object.defineProperty(newObj, key, desc);
          } else {
            newObj[key] = obj[key];
          }
        }
      }
    }
    newObj['default'] = obj;
    return newObj;
  }
}

function _typeof(obj) {
  if (typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol') {
    _typeof = function _typeof(obj) {
      return typeof obj;
    };
  } else {
    _typeof = function _typeof(obj) {
      return obj && typeof Symbol === 'function' && obj.constructor === Symbol && obj !== Symbol.prototype
        ? 'symbol'
        : typeof obj;
    };
  }
  return _typeof(obj);
}

function _toConsumableArray(arr) {
  return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _nonIterableSpread();
}

function _nonIterableSpread() {
  throw new TypeError('Invalid attempt to spread non-iterable instance');
}

function _iterableToArray(iter) {
  if (Symbol.iterator in Object(iter) || Object.prototype.toString.call(iter) === '[object Arguments]')
    return Array.from(iter);
}

function _arrayWithoutHoles(arr) {
  if (Array.isArray(arr)) {
    for (var i = 0, arr2 = new Array(arr.length); i < arr.length; i++) {
      arr2[i] = arr[i];
    }
    return arr2;
  }
}

function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError('Cannot call a class as a function');
  }
}

function _defineProperties(target, props) {
  for (var i = 0; i < props.length; i++) {
    var descriptor = props[i];
    descriptor.enumerable = descriptor.enumerable || false;
    descriptor.configurable = true;
    if ('value' in descriptor) descriptor.writable = true;
    Object.defineProperty(target, descriptor.key, descriptor);
  }
}

function _createClass(Constructor, protoProps, staticProps) {
  if (protoProps) _defineProperties(Constructor.prototype, protoProps);
  if (staticProps) _defineProperties(Constructor, staticProps);
  return Constructor;
}

function _possibleConstructorReturn(self, call) {
  if (call && (_typeof(call) === 'object' || typeof call === 'function')) {
    return call;
  }
  return _assertThisInitialized(self);
}

function _assertThisInitialized(self) {
  if (self === void 0) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }
  return self;
}

function _getPrototypeOf(o) {
  _getPrototypeOf = Object.setPrototypeOf
    ? Object.getPrototypeOf
    : function _getPrototypeOf(o) {
        return o.__proto__ || Object.getPrototypeOf(o);
      };
  return _getPrototypeOf(o);
}

function _inherits(subClass, superClass) {
  if (typeof superClass !== 'function' && superClass !== null) {
    throw new TypeError('Super expression must either be null or a function');
  }
  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {value: subClass, writable: true, configurable: true}
  });
  if (superClass) _setPrototypeOf(subClass, superClass);
}

function _setPrototypeOf(o, p) {
  _setPrototypeOf =
    Object.setPrototypeOf ||
    function _setPrototypeOf(o, p) {
      o.__proto__ = p;
      return o;
    };
  return _setPrototypeOf(o, p);
}

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {value: value, enumerable: true, configurable: true, writable: true});
  } else {
    obj[key] = value;
  }
  return obj;
}

var LazyIframe =
  /*#__PURE__*/
  (function(_Component) {
    _inherits(LazyIframe, _Component);

    function LazyIframe() {
      _classCallCheck(this, LazyIframe);

      return _possibleConstructorReturn(this, _getPrototypeOf(LazyIframe).apply(this, arguments));
    }

    _createClass(LazyIframe, [
      {
        key: 'componentDidMount',
        value: function componentDidMount() {
          _ReactLazyMedia.ReactLazyMedia.getInstance();
        }
      },
      {
        key: 'render',
        value: function render() {
          var _this$props = this.props,
            src = _this$props.src,
            classes = _this$props.classes,
            id = _this$props.id;
          return _react['default'].createElement('iframe', {
            className: _classnames['default'].apply(void 0, ['lazy'].concat(_toConsumableArray(classes))),
            id: id,
            'data-src': src
          });
        }
      }
    ]);

    return LazyIframe;
  })(_react.Component);

exports.LazyIframe = LazyIframe;

_defineProperty(LazyIframe, 'propTypes', {
  classes: _propTypes['default'].arrayOf(_propTypes['default'].string),
  id: _propTypes['default'].string,
  src: _propTypes['default'].string.isRequired
});

_defineProperty(LazyIframe, 'defaultProps', {
  classes: []
});
