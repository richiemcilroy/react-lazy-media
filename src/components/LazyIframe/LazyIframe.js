import React, {Component} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import {ReactLazyMedia} from '../../ReactLazyMedia';

export class LazyIframe extends Component {
  static propTypes = {
    classes: PropTypes.arrayOf(PropTypes.string),
    id: PropTypes.string,
    src: PropTypes.string.isRequired
  };

  static defaultProps = {
    classes: []
  };

  componentDidMount() {
    ReactLazyMedia.getInstance();
  }

  render() {
    const {src, classes, id} = this.props;
    return <iframe className={classNames('lazy', ...classes)} id={id} data-src={src}></iframe>;
  }
}
