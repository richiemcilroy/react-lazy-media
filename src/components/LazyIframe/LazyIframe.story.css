.lazyIframe {
  margin-top: 150vh;
}
