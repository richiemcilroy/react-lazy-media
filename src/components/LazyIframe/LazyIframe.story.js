import React from 'react';
import {storiesOf} from '@storybook/react';
import {withKnobs, text} from '@storybook/addon-knobs';
import {LazyIframe} from './LazyIframe';
import './LazyIframe.story.css';

storiesOf('Lazy', module)
  .addDecorator(withKnobs)
  .add('LazyIframe', () => {
    return (
      <LazyIframe
        classes={['lazyIframe']}
        src={
          'https://www.openstreetmap.org/export/embed.html?bbox=-0.004017949104309083%2C51.47612752641776%2C0.00030577182769775396%2C51.478569861898606&layer=mapnik'
        }
      />
    );
  });
