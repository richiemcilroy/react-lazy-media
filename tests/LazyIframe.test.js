import React from 'react';
import renderer from 'react-test-renderer';
import {LazyIframe} from '../src/components/LazyIframe/LazyIframe';

test('LazyIframe should require a src prop', () => {
  expect(() => {
    renderer.create(<LazyIframe />);
  }).toThrowErrorMatchingSnapshot();
});

test('LazyIframe should have default props', () => {
  const component = renderer.create(<LazyIframe src={'/page.html'} />);

  const tree = component.toJSON();

  expect(tree.props).toEqual({
    'data-src': '/page.html',
    className: 'lazy'
  });
});

test('LazyIframe should render a iframe with a src', () => {
  const component = renderer.create(<LazyIframe src={'/page.html'} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LazyIframe should have an id', () => {
  const component = renderer.create(<LazyIframe src={'/page.html'} id={'lazyIframe'} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LazyIframe should have classes', () => {
  const component = renderer.create(<LazyIframe src={'/page.html'} classes={['class1', 'class2']} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
