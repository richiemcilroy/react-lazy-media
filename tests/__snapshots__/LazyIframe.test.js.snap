// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyIframe should have an id 1`] = `
<iframe
  className="lazy"
  data-src="/page.html"
  id="lazyIframe"
/>
`;

exports[`LazyIframe should have classes 1`] = `
<iframe
  className="lazy class1 class2"
  data-src="/page.html"
/>
`;

exports[`LazyIframe should render a iframe with a src 1`] = `
<iframe
  className="lazy"
  data-src="/page.html"
/>
`;

exports[`LazyIframe should require a src prop 1`] = `
"Warning: Failed prop type: The prop \`src\` is marked as required in \`LazyIframe\`, but its value is \`undefined\`.
    in LazyIframe"
`;
